transports/vm/src/main/java/org/mule/transport/vm/VMMessageReceiver.java

```diff
@@ -197,21 +201,24 @@ public MuleEvent process() throws Exception
         int batchSize = getBatchSize(queue.size());
 
         // try to get the first event off the queue
-        MuleEvent message = (MuleEvent) queue.poll(connector.getQueueTimeout());
+        MuleMessage message = (MuleMessage) queue.poll(connector.getQueueTimeout());
```

Comment:
This poll assumes no events are left in a persistent queue when upgrading to this version.